src/machine/state/gpr.rs

```diff
@@ -2,23 +2,23 @@ use machine::state::types::Octa;
 
 use std::ops::{Index, IndexMut};
 
-pub struct GPRegisters(());
+pub struct GPRegisters(Vec<Octa>);
```

Comment:
I was using a struct instead of a tuple struct in the memory implementation. Do we want to have that uniform?